### PR TITLE
chapter 4 android-kotlin created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
* [AP - Chapter 4] Managing UI state - https://androidtraining.atlassian.net/browse/AT-46

## Description:
Chapter 4 in kotlin created

## :camera_flash: UI preview
ezgif com-video-to-gif(5)

## :pushpin: Notes:
It also contains the integration with detekt

### CODE REVIEW

File: app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        if (BuildConfig.DEBUG) {
+            Timber.plant(Timber.DebugTree())
+        }
+        Timber.d("onCreate(Bundle) called")
+        setContentView(R.layout.activity_quiz)
+
+        val currentIndex = savedInstanceState?.getInt(KEY_INDEX, 0) ?: 0
+        quizViewModel.currentIndex = currentIndex
+
+        questionTextView!!.setOnClickListener { v ->
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt]: remove the `!!`

File: app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    public override fun onDestroy() {
+        super.onDestroy()
+        Timber.d("onDestroy() called")
+    }
+
+    private fun updateQuestion() {
+        val actualQuestion = quizViewModel.currentQuestion
+        questionTextView!!.setText(quizViewModel.currentQuestionText)
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt]: remove `!!`

File: app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    private fun showToastCorrectly(isCorrect: Boolean) {
+        @StringRes
+        val answerTest = if (isCheater) R.string.judgment_toast else getAnswerText(isCorrect)
+        val toast = Toast.makeText(this, answerTest, Toast.LENGTH_SHORT)
+        toast.setGravity(Gravity.TOP or Gravity.CENTER_HORIZONTAL, 0, 0)
+        toast.show()
+    }
+
+    @StringRes
+    private fun getAnswerText(isCorrect: Boolean): Int {
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt]: use `=` intead of `{`

File: app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    private fun showPercentage() {
+        if (countOfQuestionsResolved == quizViewModel.questionBank.size) {
+            val percentage = "$countOfCorrectQuestionsResolved/$countOfQuestionsResolved"
+            Toast.makeText(this, "The average is: $percentage", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    companion object {
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt]: we usually put companion object at the beginning of the class, move it to the top

**Comment by FerrariBruno** [comment]
Should we untrack the .idea files?

**Comment by mirland** [comment]
Remove the tracked .idea files

File: build/reports/detekt/detekt.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+TooManyFunctions - 16/11 - [QuizActivity] at /home/toshiba/Desktop/Repo/app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt:24:1 - Signature=QuizActivity.kt$QuizActivity : AppCompatActivity
+NoBlankLineBeforeRbrace - [PsiWhiteSpace] at /home/toshiba/Desktop/Repo/app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt:178:5 - Signature=
```
Comment (f7deleon) [inline build/reports/detekt/detekt.txt]: maybe you should ignore this file too

File: build/reports/detekt/detekt.html [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+<ul>
+<li>
+<span class="location">
+/home/toshiba/Desktop/Repo/app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt:153:68
+</span>
+<br />
+<span class="message">
+Needless blank line(s)
+</span>
+</li>
```
Comment (f7deleon) [inline build/reports/detekt/detekt.html]: and this

File: build/tmp/detekt/detekt.args [hunk 7]
```diff
@@ -0,0 +1,11 @@
+--input
+/home/toshiba/Desktop/Repo/app/src/main/java/com/bignerdranch/android/geoquiz/models/QuizViewModel.kt,/home/toshiba/Desktop/Repo/app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt
+--config
```
Comment (f7deleon) [inline build/tmp/detekt/detekt.args]: and this

File: build/tmp/detekt/detektGenerateConfig.args [hunk 8]
```diff
@@ -0,0 +1,5 @@
+--generate-config
+--config
+/home/toshiba/Desktop/Repo/config/detekt/detekt.yml
```
Comment (f7deleon) [inline build/tmp/detekt/detektGenerateConfig.args]: probably all the build folder should be ignored

**Review by f7deleon — APPROVED** [review]

**Pull request merged**